src/scenic/syntax/compiler.py

```diff
@@ -171,3 +172,158 @@ def visit_ApparentlyFacingSpecifier(self, node: s.ApparentlyFacingSpecifier):
+            keywords=[]
+            if node.base is None
+            else [ast.keyword(arg="Y", value=self.visit(node.base))],
+        )
+
+    def visit_RelativeHeadingOp(self, node: s.RelativeHeadingOp):
+        return ast.Call(
+            func=ast.Name(id="RelativeHeading", ctx=loadCtx),
+            args=[self.visit(node.target)],
+            keywords=[]
+            if node.base is None
+            else [ast.keyword(arg="Y", value=self.visit(node.base))],
+        )
+
+    def visit_ApparentHeadingOp(self, node: s.ApparentHeadingOp):
+        return ast.Call(
+            func=ast.Name(id="ApparentHeading", ctx=loadCtx),
+            args=[self.visit(node.target)],
+            keywords=[]
+            if node.base is None
+            else [ast.keyword(arg="Y", value=self.visit(node.base))],
+        )
+
+    def visit_DistanceFromOp(self, node: s.DistanceFromOp):
+        assert node.target is not None or node.base is not None
+        target = ego if node.target is None else self.visit(node.target)
+        base = ego if node.base is None else self.visit(node.base)
+        return ast.Call(
+            func=ast.Name(id="DistanceFrom", ctx=loadCtx),
+            args=[target],
+            keywords=[ast.keyword(arg="Y", value=base)],
+        )
+
+    def visit_DistancePastOp(self, node: s.DistancePastOp):
+        return ast.Call(
+            func=ast.Name(id="DistancePast", ctx=loadCtx),
+            args=[self.visit(node.target)],
+            keywords=[]
+            if node.base is None
+            else [ast.keyword(arg="Y", value=self.visit(node.base))],
+        )
+
+    def visit_AngleFromOp(self, node: s.AngleFromOp):
+        assert (
+            node.base is not None or node.target is not None
+        ), "neither target nor base were specified in AngleFromOp"
+        base = ego if node.base is None else self.visit(node.base)
+        target = ego if node.target is None else self.visit(node.target)
+        return ast.Call(
+            func=ast.Name(id="AngleFrom", ctx=loadCtx),
+            args=[base, target],
+            keywords=[],
+        )
+
+    def visit_FollowOp(self, node: s.FollowOp):
+        return ast.Call(
+            func=ast.Name(id="Follow", ctx=loadCtx),
+            args=[
+                self.visit(node.target),
+                self.visit(node.base),
+                self.visit(node.distance),
+            ],
+            keywords=[],
+        )
+
+    def visit_VisibleOp(self, node: s.VisibleOp):
+        return ast.Call(
+            func=ast.Name(id="Visible", ctx=loadCtx),
+            args=[self.visit(node.region)],
+            keywords=[],
+        )
+
+    def visit_NotVisibleOp(self, node: s.VisibleOp):
+        return ast.Call(
+            func=ast.Name(id="NotVisible", ctx=loadCtx),
+            args=[self.visit(node.region)],
+            keywords=[],
+        )
+
+    def visit_PositionOfOp(self, node: s.PositionOfOp):
+        classToFunctionName = {
```

Comment:
Rather than have this table, which would need to be kept in sync with the node definitions, what about adding a `functionName` attribute to the node classes? That would then only need to be kept in sync with the veneer. (And we could put in a sanity check to make sure the functions exist there, like we currently do in `translator.py`.)

Reply:
fixed in c2e86eb0a062f0663f70b2675800ec0d94450ea9